src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -322,16 +333,21 @@ getCompletions uri prefixInfo (WithSnippets withSnippets) = pluginGetFile "getCo
       mkImportCompl label = (J.detail ?~ label)
                           . mkModCompl
                           $ fromMaybe "" (T.stripPrefix enteredQual label)
-
-      filtImportCompls = [ mkImportCompl label
-                          | label <- Fuzzy.simpleFilter fullPrefix importableModules
-                          , enteredQual `T.isPrefixOf` label
-                          ]
-    in return $ IdeResultOk $
-      if "import " `T.isPrefixOf` fullLine
-      then filtImportCompls
-      else filtModNameCompls ++ map (toggleSnippets . mkCompl) filtCompls
-
+      
+      filtListWith f list = [ f label
+                             | label <- Fuzzy.simpleFilter fullPrefix list
+                             , enteredQual `T.isPrefixOf` label
+                             ]
+
+      filtImportCompls = filtListWith mkImportCompl importableModules
+      filtExtensionCompls = filtListWith mkExtCompl cachedExtensions
+
+      result
+        | "import " `T.isPrefixOf` fullLine      = filtImportCompls
+        | "{-# LANGUAGE" `T.isPrefixOf` fullLine = filtExtensionCompls
```

Comment:
This is fine for most use cases, but technically the keyword `LANGUAGE` is parsed to be case insensitive.